IdentityCore/src/workplacejoin/mac/MSIDWorkPlaceJoinUtil.m

```diff
@@ -158,4 +197,65 @@ + (SecIdentityRef)copyWPJIdentity:(__unused id<MSIDRequestContext>)context
     return identityRef; //Caller must call CFRelease
 }
 
+#pragma mark - Lookup by private key
+
++ (MSIDRegistrationInformation *)findDefaultWPJRegistrationInfoWithContext:(id<MSIDRequestContext>)context
+{
+    OSStatus status = noErr;
+    CFTypeRef privateKeyCFDict = NULL;
+    
+    // Set the private key query dictionary.
+    NSDictionary *queryCert = @{ (__bridge id)kSecClass : (__bridge id)kSecClassKey,
+                                 (__bridge id)kSecAttrApplicationTag: kMSIDPrivateKeyIdentifier,
+                                 (__bridge id)kSecReturnAttributes: @YES,
+                                 (__bridge id)kSecReturnRef: @YES};
+    
+    // Get the certificate. If the certificate is not found, this is not considered an error.
+    status = SecItemCopyMatching((__bridge CFDictionaryRef)queryCert, (CFTypeRef*)&privateKeyCFDict);
+    if (status != errSecSuccess)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to find workplace join private key with status %ld", (long)status);
+        return nil;
+    }
+        
+    NSDictionary *privateKeyDict = CFBridgingRelease(privateKeyCFDict);
+    
+    /*
+     kSecAttrApplicationLabel
+     For asymmetric keys this holds the public key hash which allows digital identity formation (to form a digital identity, this value must match the kSecAttrPublicKeyHash ('pkhh') attribute of the certificate)
+     */
+    NSData *applicationLabel = privateKeyDict[(__bridge id)kSecAttrApplicationLabel];
+
+    if (!applicationLabel)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Unexpected key found without application label. Aborting lookup");
+        return nil;
+    }
+    
+    SecKeyRef privateKeyRef = (SecKeyRef)CFBridgingRetain(privateKeyDict[(__bridge id)kSecValueRef]);
```

Comment:
This method retains privateKeyRef but the method name does not suggest that. How will the caller know that it needs to release these refs.

Reply:
It returns MSIDRegistrationInformation that doesn't need to be released. But I agree that it is confusing, and I think we don't even need to retain it. So I'll update to not retain private key in both functions.

Reply:
refactored the code :)